src/main/java/com/google/cloud/spanner/pgadapter/utils/MutationWriter.java

```diff
@@ -224,17 +224,11 @@ public void writeCopyDataToErrorFile(byte[] payload) throws IOException {
   }
 
   /**
-   * The list of mutaitons will be written to the error file if a problem was encountered while
-   * generating the mutaiton list or if Spanner returns an error upon commiting the mutations.
+   * Copy data will be written to an error file if a problem was encountered while generating the
+   * mutaiton list or if Spanner returns an error upon commiting the mutations.
    */
-  public void writeMutationsToErrorFile() throws IOException {
-    if (this.fileWriter == null) {
-      createErrorFile();
-    }
-
-    for (Mutation mutation : this.mutations) {
-      this.fileWriter.write(mutation.toString());
-    }
+  public void writeCopyDataToErrorFile() throws IOException {
```

Comment:
I think every time we write to the error file we write the full payload so you can just have a single  version  of writeCopyDataToErrorFile() without having a payload argument

Reply:
Done.